### DISCUSSION TITLE
Update row in Datagrid

### DISCUSSION THREAD

**Discussion by garybushey** [microsoft/fluentui#28704] [Q&A]
This should be easy but I cannot get this to work.  I want to update a cell in my DataGrid (using Typescript).  I have a variable called "ruleTemplates" that I have set to my DataGrid's "items" parameter.  Information is loading just fine.   What I want to do is update the "status" field in one row but I cannot seem to get it to work. I have tried updating but "ruleTemplates" but the UI doesn't show any updates.  Any ideas?

**Comment by logicaloud** [comment]
I assume your `ruleTemplates` are in a state variable? I.e., along those lines:

```
const [ruleTemplates, setRuleTemplates] = useState([]);
```

Then you could update a cell in the ruleTemplates array by constructing a new array with one element replaced, i.e., something like:

```
let newRuleTemplates = [...ruleTemplates]; // clone array
newRuleTemplates[x] = newItemWithNewStatus; // set new item with updated status
setRuleTemplates(newRuleTemplates); // set state
```

See also: https://react.dev/learn/updating-arrays-in-state#replacing-items-in-an-array

Does that help?
